### PR TITLE
Offscreen invalidate and sizing fix

### PR DESCRIPTION
**Pull request by gerhardberger**
This PR adds an `invalidate` method to `webContents` to manually request a new frame in offscreen mode.

And also fixes the issue that a browser did not resize properly in offscreen mode.

### CODE REVIEW

**Comment by zeke** [comment]
Hi @gerhardberger. What is the use case for this new method?

**Comment by gerhardberger** [comment]
@zeke, it generates a `'paint'` event on-demand with the current frame. This can be used when you need the latest frame but no change happens on the page, e.g. you start rendering to a new place and you want an initial frame.

File: atom/browser/native_window.cc [hunk 1]
```diff
@@ -204,6 +204,14 @@ void NativeWindow::InitFromOptions(const mate::Dictionary& options) {
 
 void NativeWindow::SetSize(const gfx::Size& size, bool animate) {
   SetBounds(gfx::Rect(GetPosition(), size), animate);
+
+  // In offscreen mode call RenderWidgetHostView's SetSize explicitly
+  const auto view = web_contents()->GetRenderWidgetHostView();
+  if (view) {
```
Comment (zcbenz) [inline atom/browser/native_window.cc]: `IsOffScreen` should be checked here.

Comment (zcbenz) [inline atom/browser/native_window.cc]: How about subscribing to window resize events in `RenderWidgetHostView`? It is provided by `NativeWindowObserver`.

Comment (gerhardberger) [inline atom/browser/native_window.cc]: Great I didn't notice that, thanks! I moved the call to `OffscreenRenderWidgetHostView`.

**Comment by danburzo** [comment]
This is great! I was doing a `stopPainting()/startPainting()` combo to force a `paint` event. 

The use-case (for me, at least) is you want to grab a PNG representation of the page _after_ a certain client event is emitted, and since the Buffer is volatile, I'm not sure how you'd have done it previously (other than triggering a paint event).

**Pull request merged**